### ISSUE TITLE
Incorrect target file for gen-version in the failure() function

### ISSUE THREAD

**Issue by ahmkar94** [kernelkit/infix#562]
### Current Behavior

In the failure() function, the gen-version command is currently writing its output to the wrong directory. The command:
```
gen-version >"$FACTORY_D/20-version.json"
```
is targeting the $FACTORY_D directory. 

### Expected Behavior

It should instead be targeting the $FAILURE_D directory, like the other commands in this function.
```
gen-version >"$FAILURE_D/20-version.json"
```

**Comment by troglobit** [comment]
Nice catch, confirmed!

**Comment by troglobit** [comment]
We didn't cover this one in today's inbox meeting, but I'm assigning this one to you. Just make sure to include the "Fixes #562" line somewhere in the commit message for this fix :sunglasses:

**Comment by ahmkar94** [comment]
> We didn't cover this one in today's inbox meeting, but I'm assigning this one to you, @ahmkar94. Just make sure to include the "Fixes #562" line somewhere in the commit message for this fix 😎

Did that yesterday in the ongoing PR :)
